src/database/database.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <iostream>
+#include <cstdint>
+#include <vector>
+
+#include <mongocxx/client.hpp>
```

Comment:
If you want to use these classes you need to link mongo library `PUBLIC`ly so the `"main"` file knows where to find them when `database.hpp` is included.
You could also `forward declare` them and make them `unique_ptr`s as we do not care about heap usage and it will make a `hpp` a bit cleaner.
Whatever you choose is fine.